### PR TITLE
chore: restore fetching framework version from Bundle IC-23

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/IC-23" title="IC-23" target="_blank">TaskIC-23</a>  Find alternative solution for injecting DM and SE framework versions
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

This PR restores the previous behavior regarding fetching the framework version from the bundle. The workaround to hardcode the version was only temporary and is already out of date. Let's revisit this issue again before the next release.

See https://github.com/wireapp/wire-ios-data-model/pull/1309.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Source/Synchronization/ZMHotFix.m [hunk 1]
```diff
@@ -61,8 +61,7 @@ - (void)applyPatches
         return;
     }
     
-   /// Normally we fetch the short version from the framework bundle, but it's not working so we hardcode the version.
-    NSString * currentVersionString = @"402.0.0";
+    NSString * currentVersionString = [[[NSBundle bundleForClass:ZMUserSession.class] infoDictionary] objectForKey:@"CFBundleShortVersionString"];
```
Comment (sb88k) [inline Source/Synchronization/ZMHotFix.m:64]: Mean to try `Self.self` in Swift? just like in data model.

Not sure the equivalent in objc, maybe [self class]?

**Review by KaterinaWire — APPROVED** [review]

File: Source/Synchronization/ZMHotFix.m (hunk 1 above)
Comment (johnxnguyen) [reply Source/Synchronization/ZMHotFix.m:64]: I don't think there is an equivalent, but I'm also not sure if there was an issue here to begin with. I thought I'd try to revert it back to the way it was before and then see if the issue still persists.

**Review by sb88k — APPROVED** [review]

**Comment by johnxnguyen** [comment]
release patch

**Comment by zenkins** [comment]
Hello, @johnxnguyen! Merging PR for command **release patch** for PR #**1616**. Will create a **patch** release after merged.
Next version is 405.0.3.

**Pull request merged**